internal/evaluator/conftest_evaluator_test.go

```diff
@@ -1235,6 +1248,45 @@ func TestRuleMetadata(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "drop stale effectiveOn",
+			result: output.Result{
+				Metadata: map[string]any{
+					"code":        "warning2",
+					"collections": []any{"A"},
+				},
+			},
+			rules: rules,
+			code:  "warning2",
+			want: output.Result{
+				Metadata: map[string]any{
+					"code":        "warning2",
+					"collections": []string{"A"},
+					"description": "Warning 2 description",
+					"title":       "Warning2",
+				},
+			},
+		},
+		{
+			name: "add relevant effectiveOn",
+			result: output.Result{
+				Metadata: map[string]any{
+					"code":        "warning3",
+					"collections": []any{"A"},
+				},
+			},
+			rules: rules,
+			code:  "warning3",
+			want: output.Result{
+				Metadata: map[string]any{
+					"code":         "warning3",
+					"collections":  []string{"A"},
+					"description":  "Warning 3 description",
+					"title":        "Warning3",
+					"effective_on": effectiveOnTest,
+				},
+			},
+		},
```

Comment:
Looks good.

Reply:
For the record I deleted a comment about using `[]string{"A"},` instead of `[]any{"A"},` after looking a bit more carefully.